src/Rings/groebner.jl

```diff
@@ -1048,6 +1080,13 @@ function normal_form(A::Vector{T}, J::MPolyIdeal; ordering::MonomialOrdering=def
   return res
 end
 
+function is_normal_form_f4_applicable(I::MPolyIdeal, ordering::MonomialOrdering)
+    return (ordering == degrevlex(base_ring(I)) && !is_graded(base_ring(I))
+            && ((coefficient_ring(I) isa FqField
+                 && absolute_degree(coefficient_ring(I)) == 1
+                 && characteristic(coefficient_ring(I)) < 2^31)))
```

Comment:
Just wondering: `is_f4_applicable` also has `|| coefficient_ring(I) == QQ)` in the condition. I assume this one is intentionally different, right?

Reply:
Yes, that's intentionally different, for normal forms we still need to implement the lifting of the results on the msolve side.